src/components/secure-backup/index.tsx

```diff
@@ -61,92 +76,102 @@ export class SecureBackup extends React.PureComponent<Properties, State> {
-      <>
-        <p>
-          The following recovery key has been generated for you. Click `Copy` to copy it to your clipboard. Store it in
-          a secure place such as a password manager of vault then click `Save Backup` to complete the backup process.
-        </p>
-        {this.props.recoveryKey && <div {...cn('recovery-key')}>{this.props.recoveryKey}</div>}
-        {this.renderButtons(
+  renderBackupContent = () => {
+    const {
+      backupStage,
+      onGenerate,
+      onRestore,
+      onVerifyKey,
+      onClose,
+      onSave,
+      recoveryKey,
+      errorMessage,
+      successMessage,
+    } = this.props;
+
+    switch (backupStage) {
+      case BackupStage.None:
+      case BackupStage.SystemPrompt:
+        const promptProps = { ...(this.isSystemPrompt && { onClose }) };
+        return (
           <>
-            {this.props.recoveryKey && <Button onPress={this.copyKey}>Copy</Button>}
-            <Button onPress={() => this.props.onSave()} isDisabled={!this.state.hasCopied}>
-              Save backup
-            </Button>
-          </>
-        )}
-      </>
-    );
-  }
+            {this.noBackupExists && (
+              <GeneratePrompt isSystemPrompt={this.isSystemPrompt} onGenerate={onGenerate} {...promptProps} />
+            )}
 
-  renderRestoreMode() {
-    return (
-      <>
-        <Input
-          placeholder='Enter your recovery key'
-          onChange={this.trackRecoveryKey}
-          value={this.state.userInputRecoveryKey}
-        />
-        {this.renderButtons(
-          <Button onPress={() => this.props.onRestore(this.state.userInputRecoveryKey)}>Restore backup</Button>
-        )}
-      </>
-    );
-  }
+            {this.backupNotRestored && (
+              <RestorePrompt isSystemPrompt={this.isSystemPrompt} onVerifyKey={onVerifyKey} {...promptProps} />
+            )}
 
-  renderButtons = (buttons) => {
-    return (
-      <>
-        <div {...cn('messages')}>
-          {this.props.successMessage && <Alert variant='success'>{this.props.successMessage}</Alert>}
-          {this.props.errorMessage && <Alert variant='error'>{this.props.errorMessage}</Alert>}
-        </div>
-        <div {...cn('footer')}>{buttons}</div>
-      </>
-    );
+            {this.isRecovered && (
+              <RecoveredBackup onClose={onClose} onGenerate={this.props.onGenerate} isLegacy={this.props.isLegacy} />
+            )}
+          </>
+        );
+
+      case BackupStage.GenerateBackup:
+        return (
+          <GenerateBackup
+            recoveryKey={recoveryKey}
+            errorMessage={errorMessage}
+            hasCopied={this.state.hasCopied}
+            onCopy={this.copyKey}
+            onSave={onSave}
+          />
+        );
+
+      case BackupStage.RestoreBackup:
+        return (
+          <RestoreBackup
+            userInputRecoveryKey={this.state.userInputRecoveryKey}
+            errorMessage={errorMessage}
+            trackRecoveryKey={this.trackRecoveryKey}
```

Comment:
This feels specific to the `RestoreBackup` component. Push it down to that level.